catalyst-gateway/bin/src/service/common/types/cardano/ada_value.rs

```diff
@@ -39,20 +39,19 @@ static SCHEMA: LazyLock<MetaSchema> = LazyLock::new(|| {
 
 pub(crate) struct AdaValue(u64);
 
-impl Deref for AdaValue {
-    type Target = u64;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
-    }
-}
-
 impl Display for AdaValue {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         write!(f, "{}", self.0)
     }
 }
 
+impl AdaValue {
+    /// Performs safe addition that returns None instead of wrapping around on overflow.
+    pub(crate) fn checked_add(self, v: Self) -> Option<Self> {
```

Comment:
I would just use saturation here.  And log an error if it happens.

It's realistically impossible because there are not u64 total Lovelace in the system .  So even if you held all ADA you could not saturate a u64.

Otherwise your forcing other code to handle an error that's actually impossible which just makes that code less clear.